packages/attachments/src/AttachmentQueue.ts

```diff
@@ -0,0 +1,250 @@
+import { AbstractPowerSyncDatabase, ILogger } from '@powersync/common';
+import { AttachmentContext } from './AttachmentContext.js';
+import { LocalStorageAdapter } from './LocalStorageAdapter.js';
+import { RemoteStorageAdapter } from './RemoteStorageAdapter.js';
+import { ATTACHMENT_TABLE, AttachmentRecord, AttachmentState } from './Schema.js';
+import { StorageService } from './StorageService.js';
+import { WatchedAttachmentItem } from './WatchedAttachmentItem.js';
+
+export class AttachmentQueue {
+  periodicSyncTimer?: ReturnType<typeof setInterval>;
+  context: AttachmentContext;
+  storageService: StorageService;
+  localStorage: LocalStorageAdapter;
+  remoteStorage: RemoteStorageAdapter;
+  attachmentsDirectory?: string;
+  tableName?: string;
+  logger?: ILogger;
+  syncInterval: number = 30 * 1000;
+  syncThrottleDuration: number;
+  downloadAttachments: boolean = true;
+  watchActiveAbortController?: AbortController;
+  archivedCacheLimit: number;
+
+  constructor({
+    db,
+    localStorage,
+    remoteStorage,
+    watchAttachments,
+    logger,
+    tableName = ATTACHMENT_TABLE,
+    syncInterval = 30 * 1000,
```

Comment:
We typically indicate the millisecond unit in the variable name
```suggestion
    syncIntervalMs= 30 * 1000,
```